compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1934,26 +1938,29 @@ private BLangSimpleVariable defVariable(Location pos, BType varType, BLangBlockS
     private void createAndAddBoundVariableDef(BLangBlockStmt parentBlockStmt,
                                               BLangErrorVariable.BLangErrorDetailEntry detailEntry,
                                               BLangExpression detailEntryVar) {
-        if (detailEntry.valueBindingPattern.getKind() == NodeKind.VARIABLE) {
-            BLangSimpleVariableDef errorDetailVar = createVarDef(
-                    ((BLangSimpleVariable) detailEntry.valueBindingPattern).name.value,
-                    detailEntry.valueBindingPattern.type,
-                    detailEntryVar,
-                    detailEntry.valueBindingPattern.pos);
-            parentBlockStmt.addStatement(errorDetailVar);
-
-        } else if (detailEntry.valueBindingPattern.getKind() == NodeKind.RECORD_VARIABLE) {
-            BLangRecordVariableDef recordVariableDef = ASTBuilderUtil.createRecordVariableDef(
-                    detailEntry.valueBindingPattern.pos,
-                    (BLangRecordVariable) detailEntry.valueBindingPattern);
-            recordVariableDef.var.expr = detailEntryVar;
-            recordVariableDef.type = symTable.recordType;
-            parentBlockStmt.addStatement(recordVariableDef);
+        BLangVariable valueBindingPattern = detailEntry.valueBindingPattern;
+        NodeKind valueBindingPatternKind = valueBindingPattern.getKind();
 
-        } else if (detailEntry.valueBindingPattern.getKind() == NodeKind.TUPLE_VARIABLE) {
-            BLangTupleVariableDef tupleVariableDef = ASTBuilderUtil.createTupleVariableDef(
-                    detailEntry.valueBindingPattern.pos, (BLangTupleVariable) detailEntry.valueBindingPattern);
-            parentBlockStmt.addStatement(tupleVariableDef);
+        if (valueBindingPatternKind == NodeKind.VARIABLE) {
+            BLangSimpleVariableDef errorDetailVar = createVarDef(((BLangSimpleVariable) valueBindingPattern).name.value,
+                    valueBindingPattern.type, detailEntryVar, valueBindingPattern.pos);
+            parentBlockStmt.addStatement(errorDetailVar);
+        } else {
+            valueBindingPattern.expr = detailEntryVar;
+            BLangNode blockStatementNode = rewrite(valueBindingPattern, env);
+            List<BLangStatement> statements = ((BLangBlockStmt) blockStatementNode).stmts;
+            for (int i = 0; i < statements.size(); i++) {
+                BLangStatement bLangStatement = statements.get(i);
+                if (i == 0) {
+                    // Wrap first virtual var def for init expression with a block statement to add it directly to
+                    // init function in module var case
+                    BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(bLangStatement.pos);
+                    blockStmt.addStatement(bLangStatement);
+                    parentBlockStmt.addStatement(blockStmt);
+                } else {
+                    parentBlockStmt.addStatement(bLangStatement);
+                }
+            }
```

Comment:
IMO we can improve this to avoid having to check `i ==0` for each statement.

Something like
```suggestion
            List<BLangStatement> statements = ((BLangBlockStmt) blockStatementNode).stmts;
            if (statements.isEmpty()) { // Not sure if we need this (if statements can be empty here).
                return;
            }

            // Wrap first virtual var def for init expression with a block statement to add it directly to
            // init function in module var case.
            BLangStatement bLangStatement = statements.get(0);
            BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(bLangStatement.pos);
            blockStmt.addStatement(bLangStatement);
            parentBlockStmt.addStatement(blockStmt);

            for (int i = 1; i < statements.size(); i++) {
                parentBlockStmt.addStatement(statements.get(i));
            }
```